### ISSUE TITLE
No connection adapters were found for magnet

### ISSUE THREAD

**Issue by a1ad** [Flexget/Flexget#3154]
Hi guys,

Same problem as:
https://github.com/Flexget/Flexget/issues/2230
https://github.com/CouchPotato/CouchPotatoServer/pull/7325
https://github.com/SickChill/SickChill/issues/5600

I am using jackett rss.

`2021-09-12 12:00:33 INFO     download      F1              Downloading: Formula.1.2021x12.Belgium.Race.SkyF1HD.1080p
2021-09-12 12:00:33 DEBUG    download      F1              Downloading url 'http://10.30.52.93:9117/dl/ettv/?jackett_apikey=ny09m6whlnh7bily2vkd9f36l6ixb2l9&path=Q2ZESjhFLVFfSUZjcU1oSmg2c3B4VTVORFdFaUJKTWpsbUs0cnZLRjhMOVRkQ3ZETVBmOTlkekdpbGJEeXlrLWhXeVJEa0xXWDdINFlaMGZZTWNzc0NjYWVSNkV4dGdvUGxsR2tpSUFPWnhmX01ocEJzanZtUm1NcFJnblBtb1RTQ0hROTRYa2xaaTE4M0JycEgtM3lsWnNxczRpUVJIcWc5OGhNc1JScEFVMmh4cDZwSlhFNVpBM1VnNGtacnBsYUdVT0R1TnNSNF94TUNTQmhzdi13dlZZTW45d21IVFVGQmxSZmNOMWlYaUl1Y1Rl&file=Formula.1.2021x12.Belgium.Race.SkyF1HD.1080p'
2021-09-12 12:00:33 DEBUG    utils.requests F1              GETing URL http://10.30.52.93:9117/dl/ettv/?jackett_apikey=ny09m6whlnh7bily2vkd9f36l6ixb2l9&path=Q2ZESjhFLVFfSUZjcU1oSmg2c3B4VTVORFdFaUJKTWpsbUs0cnZLRjhMOVRkQ3ZETVBmOTlkekdpbGJEeXlrLWhXeVJEa0xXWDdINFlaMGZZTWNzc0NjYWVSNkV4dGdvUGxsR2tpSUFPWnhmX01ocEJzanZtUm1NcFJnblBtb1RTQ0hROTRYa2xaaTE4M0JycEgtM3lsWnNxczRpUVJIcWc5OGhNc1JScEFVMmh4cDZwSlhFNVpBM1VnNGtacnBsYUdVT0R1TnNSNF94TUNTQmhzdi13dlZZTW45d21IVFVGQmxSZmNOMWlYaUl1Y1Rl&file=Formula.1.2021x12.Belgium.Race.SkyF1HD.1080p with args () and kwargs {'auth': None, 'headers': {'User-Agent': 'FlexGet/3.1.137 (www.flexget.com)', 'Accept-Encoding': 'gzip, deflate, br', 'Accept': '*/*', 'Connection': 'keep-alive'}, 'allow_redirects': True, 'timeout': 30}
2021-09-12 12:00:34 WARNING  download      F1              RequestException No connection adapters were found for 'magnet:?xt=urn:btih:3182c3156bd20bf22a9177535c715ad69a85b20a&dn=Formula.1.2021x12.Belgium.Race.SkyF1HD.1080p&xl=22168354013&tr=udp%3A%2F%2F9.rarbg.me:2720/announce&tr=udp%3A%2F%2F9.rarbg.to:2710/announce&tr=udp%3A%2F%2F9.rarbg.to:2720/announce&tr=udp%3A%2F%2Fexplodie.org:6969/announce&tr=udp%3A%2F%2Ftracker.coppersurfer.tk:6969/announce&tr=udp%3A%2F%2Ftracker.leechers-paradise.org:6969/announce&tr=udp%3A%2F%2Ftracker.opentrackr.org:1337/announce&tr=udp%3A%2F%2Ftracker.torrent.eu.org:451/announce', while downloading http://10.30.52.93:9117/dl/ettv/?jackett_apikey=ny09m6whlnh7bily2vkd9f36l6ixb2l9&path=Q2ZESjhFLVFfSUZjcU1oSmg2c3B4VTVORFdFaUJKTWpsbUs0cnZLRjhMOVRkQ3ZETVBmOTlkekdpbGJEeXlrLWhXeVJEa0xXWDdINFlaMGZZTWNzc0NjYWVSNkV4dGdvUGxsR2tpSUFPWnhmX01ocEJzanZtUm1NcFJnblBtb1RTQ0hROTRYa2xaaTE4M0JycEgtM3lsWnNxczRpUVJIcWc5OGhNc1JScEFVMmh4cDZwSlhFNVpBM1VnNGtacnBsYUdVT0R1TnNSNF94TUNTQmhzdi13dlZZTW45d21IVFVGQmxSZmNOMWlYaUl1Y1Rl&file=Formula.1.2021x12.Belgium.Race.SkyF1HD.1080p
2021-09-12 12:00:34 DEBUG    entry         F1              Marking entry 'Formula.1.2021x12.Belgium.Race.SkyF1HD.1080p' as failed
2021-09-12 12:00:34 ERROR    entry         F1              Failed Formula.1.2021x12.Belgium.Race.SkyF1HD.1080p (Network error during request: No connection adapters were found for 'magnet:?xt=urn:btih:3182c3156bd20bf22a9177535c715ad69a85b20a&dn=Formula.1.2021x12.Belgium.Race.SkyF1HD.1080p&xl=22168354013&tr=udp%3A%2F%2F9.rarbg.me:2720/announce&tr=udp%3A%2F%2F9.rarbg.to:2710/announce&tr=udp%3A%2F%2F9.rarbg.to:2720/announce&tr=udp%3A%2F%2Fexplodie.org:6969/announce&tr=udp%3A%2F%2Ftracker.coppersurfer.tk:6969/announce&tr=udp%3A%2F%2Ftracker.leechers-paradise.org:6969/announce&tr=udp%3A%2F%2Ftracker.opentrackr.org:1337/announce&tr=udp%3A%2F%2Ftracker.torrent.eu.org:451/announce')
2021-09-12 12:00:34 VERBOSE  task          F1              FAILED: `Formula.1.2021x12.Belgium.Race.SkyF1HD.1080p` by transmission plugin because network error during request: No connection adapters were found for 'magnet:?xt=urn:btih:3182c3156bd20bf22a9177535c715ad69a85b20a&dn=Formula.1.2021x12.Belgium.Race.SkyF1HD.1080p&xl=22168354013&tr=udp%3A%2F%2F9.rarbg.me:2720/announce&tr=udp%3A%2F%2F9.rarbg.to:2710/announce&tr=udp%3A%2F%2F9.rarbg.to:2720/announce&tr=udp%3A%2F%2Fexplodie.org:6969/announce&tr=udp%3A%2F%2Ftracker.coppersurfer.tk:6969/announce&tr=udp%3A%2F%2Ftracker.leechers-paradise.org:6969/announce&tr=udp%3A%2F%2Ftracker.opentrackr.org:1337/announce&tr=udp%3A%2F%2Ftracker.torrent.eu.org:451/announce'`

The magnet link in the logs is a good one, i can add them manually in transmission.

**Comment by gazpachoking** [comment]
You are using download plugin on this task? Download plugin can't work with magnets, you have to use the transmission plugin to add the magnet directly to transmission. If you are already doing that, post your config.

**Comment by Critical-Impact** [comment]
DEBUG    utils.requests mytask       No adaptor, passing off to urllib
2022-01-14 17:07:31 ERROR    utils.requests mytask      Error getting magnet:?xt=urn:btih:magnetdatawouldbehere: <urlopen error unknown url type: magnet>

I think this has somewhat helped but I'm still not able to process magnet links from jackett. Even when I use convert_magnet it still looks like it passes it off to urllib and attempts to access the url?

**Comment by gazpachoking** [comment]
It should already be an exception though if it tries to pass to urllib from there, and break.

**Comment by Arturoe1** [comment]
I have the same issue, it seems it is not detecting the magnets . Do we have a solution for this?

**Comment by crawln45** [comment]
Libtorrent is installed with every single torrent client. Difference is Deluge is python based as suck installs the package into your systems packages. If all you need is libtorrrent you should if I remember correctly just be able to 'pip install libtorrent'

**Comment by BrutuZ** [comment]
Last I checked `libtorrent` had no pre-compiled binaries for Windows, had to manually compile it

**Comment by Karmalakas** [comment]
Yeah.. I'm not familiar with building libraries so didn't even try 🙂  Was only following some comments, making some observations and that's it. All I wanted here, is to share the solution that helped me with magnets 🙂

**Comment by diogosena** [comment]
This still a problem with jackett and qbittorrent, some magnets works, some don't

"RequestException No connection adapters were found for 'magnet:?xt=urn:btih..."

**Comment by gazpachoking** [comment]
Is this better after f57d60e6a0c601d045f8160f1bf8635bd5aa2e33?

**Comment by diogosena** [comment]
> Is this better after [f57d60e](https://github.com/Flexget/Flexget/commit/f57d60e6a0c601d045f8160f1bf8635bd5aa2e33)?

Yes, never had that problem again!
